ql/src/test/queries/clientnegative/strict_numeric_to_timestamp.q

```diff
@@ -0,0 +1,2 @@
+set hive.strict.timestamp.conversion=true;
+select cast(123 as timestamp);
```

Comment:
Can we add a couple of negative tests on column ref instead of constant? And/or with some complex expressions?

Reply:
I've added a few more tests; and I've also added one with a `struct` - however I don't really feel it that much different from the others (the way the check works) - I might be too much "inside the box"; so if you have any suggestions what tests to add - please let me know

Reply:
Maybe a couple with a comparison timestamp and numeric (e.g., `=` or `<`) which should lead to a cast to double being introduced (yeah, comparison is effectively done in double) and then the query failing.

Reply:
good catch! I didn't expected that `=` will work without casting - but since its using the converters it could avoid these checks.
I considered moving the check into the converters - however that would involved getting a `SessionState` in serde codes...so I instead added 1 more check to cover all the comparisions. Generic binary operator seem to don't auto-convert things ; `+` was complaining about adding a timestamp and an integer - so I think we are good.